src/Raven.Server/Documents/Patch/BlittableObjectInstance.cs

```diff
@@ -142,36 +143,76 @@ private JsValue GetPropertyValue(string key, int propertyIndex)
                 _parent.Blittable.GetPropertyByIndex(propertyIndex, ref propertyDetails, true);
 
                 return TranslateToJs(_parent, key, propertyDetails.Token, propertyDetails.Value);
-            }                        
+            }
+                
+            private ArrayInstance GetArrayInstanceFromBlittableArray(Engine e, BlittableJsonReaderArray bjra, BlittableObjectInstance parent)
+            {
+                ArrayInstance jsArray;
+                bjra.NoCache = true;
+                if (bjra.Length < ArrayInstance.MaxDenseArrayLength)
+                {
+                    PropertyDescriptor[] items = new PropertyDescriptor[bjra.Length];
+                    for (var i = 0; i < bjra.Length; i++)
+                    {
+                        var blit = bjra.GetValueTokenTupleByIndex(i);
+                        BlittableJsonToken itemType = blit.Item2 & BlittableJsonReaderBase.TypesMask;
+                        JsValue item;
+                        if (itemType == BlittableJsonToken.Integer || itemType == BlittableJsonToken.LazyNumber)
+                        {
+                            item = TranslateToJs(null, null, blit.Item2, blit.Item1);
+                        }
+                        else
+                        {
+                            item = TranslateToJs(parent, null, blit.Item2, blit.Item1);
+                        }
+                        items[i] = new PropertyDescriptor(item, true, true, true);
+                    }
```

Comment:
Boolean translation to flags has slight performance penalty, use flags directly when possible

Reply:
This overload is "protected internal", are you sure you want to expose it?

Reply:
I wouldn't expose to "normal" consumer, but RavenDB is quite different with separate fork :) Yes maybe can leave as-is it's really micro-optimization.